crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -388,24 +464,16 @@ impl Timeline {
     ///
     /// * `new_content` - The content of the reply
     ///
-    /// * `edit_item` - The event item you want to edit
+    /// * `edit_info` - A wrapper that contains the event ID and the content of
+    ///  the event to edit
     #[instrument(skip(self, new_content))]
     pub async fn edit(
         &self,
         new_content: RoomMessageEventContentWithoutRelation,
-        edit_item: &EventTimelineItem,
+        edit_info: EditInfo,
     ) -> Result<(), SendEventError> {
```

Comment:
This is not fallible anymore \o/ Can you get rid of the `Result`?

Reply:
we still need it because we use `self.send(content.into()).await?;`

Reply:
Oops you're right! So like the other function, this can be a plain `RoomSendQueueError` now.